WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -227,6 +227,8 @@ def post_results
     end
     post = Post.new(title: title, body: body, author: current_user, world_readable: true)
     post_post(post)
+    comp.results_posted = Time.now
+    comp.save!
```

Comment:
Good catch, @larspetrus. I also like the convention that `redirect` and `render`s come last. (I did not know that you could do them in the middle either).

I'd vote for moving the call to `post_post()` back to the end, and renaming the method to `post_post_and_redirect` (technically it also renders, but hey, in that case it didn't even actually post the post).